test/test-helpers.cpp

```diff
@@ -0,0 +1,289 @@
+
+	const auto overwrite = true;
+	::setenv(var, expected.c_str(), (int)overwrite);
+
+	char* value = ::getenv(var);
+	REQUIRE_FALSE(value == nullptr);
+	REQUIRE(expected == value);
+	value = nullptr;
+
+	{
+		TestHelpers::EnvVar envVar(var);
+
+		value = ::getenv(var);
+		REQUIRE_FALSE(value == nullptr);
+		REQUIRE(expected == value);
+		value = nullptr;
+
+		envVar.unset();
+
+		value = ::getenv(var);
+		REQUIRE(value == nullptr);
+		value = nullptr;
+	}
+
+	::unsetenv(var);
+}
+
+TEST_CASE("EnvVar::unset() doesn't change the value to which the environment "
+		"variable is restored",
+		"[test-helpers]")
+{
+	const char var[] = "nEwSb0a7-tEsT-eNvIroNm3Nt-v4rIabLe";
+	{
+		INFO("Checking that noone else is using that environment variable");
+		REQUIRE(::getenv(var) == nullptr);
+	}
+
+	const auto expected = std::string("let's try this out, shall we?");
+
+	const auto overwrite = true;
+	::setenv(var, expected.c_str(), (int)overwrite);
+
+	char* value = ::getenv(var);
+	REQUIRE_FALSE(value == nullptr);
+	REQUIRE(expected == value);
+	value = nullptr;
+
+	{
+		TestHelpers::EnvVar envVar(var);
+
+		value = ::getenv(var);
+		REQUIRE_FALSE(value == nullptr);
+		REQUIRE(expected == value);
+		value = nullptr;
+
+		envVar.unset();
+
+		value = ::getenv(var);
+		REQUIRE(value == nullptr);
+		value = nullptr;
+	}
+
+	value = ::getenv(var);
+	REQUIRE_FALSE(value == nullptr);
+	REQUIRE(expected == value);
+	value = nullptr;
+
+	::unsetenv(var);
+}
+
+TEST_CASE("EnvVar::unset() runs a function (set by on_change()) after changing "
+		"the environment variable",
+		"[test-helpers]")
+{
+	const char var[] = "nEwSb0a7-tEsT-eNvIroNm3Nt-v4rIabLe";
+	{
+		INFO("Checking that noone else is using that environment variable");
+		REQUIRE(::getenv(var) == nullptr);
+	}
+
+	const auto expected = std::string("let's try this out, shall we?");
+
+	const auto overwrite = true;
+	::setenv(var, expected.c_str(), (int)overwrite);
+
+	char* value = ::getenv(var);
+	REQUIRE_FALSE(value == nullptr);
+	REQUIRE(expected == value);
+	value = nullptr;
+
+	{
+		TestHelpers::EnvVar envVar(var);
+
+		auto counter = unsigned{};
+		auto as_expected = false;
+		envVar.on_change([&counter, &as_expected, var]() {
+			as_expected = nullptr == ::getenv(var);
+			counter++;
+		});
+
+		value = ::getenv(var);
+		REQUIRE_FALSE(value == nullptr);
+		REQUIRE(expected == value);
+		value = nullptr;
+
+		envVar.unset();
+
+		REQUIRE(as_expected);
+		REQUIRE(counter == 1);
+	}
+
+	::unsetenv(var);
+}
+
+TEST_CASE("EnvVar's destructor runs a function (set by on_change()) after "
+		"restoring the varibale to its original state",
+		"[test-helpers]")
+{
+	const char var[] = "nEwSb0a7-tEsT-eNvIroNm3Nt-v4rIabLe";
+	{
+		INFO("Checking that noone else is using that environment variable");
+		REQUIRE(::getenv(var) == nullptr);
+	}
+
+	const auto expected = std::string("let's try this out, shall we?");
+
+	SECTION("Variable wasn't even set") {
```

Comment:
Empty section?

Reply:
Yes. The point is to run this test twice, but in one run the variable is already set when `EnvVar` is created.

Catch framework runs tests in such a way that each section is entered once. So if I have two sections, the test will be run twice, entering first section on some run and the second on another.

I'll put all that in a comment.